pkg/agent/flexvolume/controller.go

```diff
@@ -116,12 +116,18 @@ func (c *FlexvolumeController) Attach(attachOpts AttachOptions, devicePath *stri
 			if index != -1 {
 				// check if the RW attachment is orphaned.
 				attachment := &volumeattachObj.Attachments[index]
+
+				logger.Infof("Volume attachment record %s/%s exists for pod: %s/%s", volumeattachObj.Namespace, volumeattachObj.Name, attachment.PodNamespace, attachment.PodName)
+				// Note this could return the reference of the pod who is requesting the attach if this pod have the same name as the pod in the attachment record.
 				pod, err := c.context.Clientset.Core().Pods(attachment.PodNamespace).Get(attachment.PodName, metav1.GetOptions{})
-				if err != nil {
-					if !errors.IsNotFound(err) {
+				if err != nil || (attachment.PodNamespace == attachOpts.PodNamespace && attachment.PodName == attachOpts.Pod) {
+					if err != nil && !errors.IsNotFound(err) {
 						return fmt.Errorf("failed to get pod CRD %s/%s. %+v", attachment.PodNamespace, attachment.PodName, err)
 					}
 
+					logger.Infof("Volume attachment record %s/%s is orphaned. Updating record with new attachment information for pod %s/%s", volumeattachObj.Namespace, volumeattachObj.Name, attachOpts.PodNamespace, attachOpts.Pod)
+
+					// Original pod doest not exist.
```

Comment:
this comment has a typo in it ("doest") and i'm not sure the comment adds much value beyond the comment on the line directly below

Reply:
I think it does add value. If a volumeattachment record exists, two outcomes can happen. 1- Volume is actually mounted by an different node and we should not proceed
2- Volume was mounted by a pod that was deleted and left orphaned. So the record should be updated.

This log is showing that the 2nd case is happening.

Ill fix the typo.

Reply:
ahh you were referring to the commented line and not to the log. Sure i can remove that